Add writing style guide (#960)
Rebilly/api-definitions: WRITING-STYLE.md

- Avoid future (often includes "will") and past tense (often includes words ending in "ed").
- Avoid [passive voice](https://developers.google.com/tech-writing/one/active-voice).
- Avoid using ("should", "could", "can").
- To add multi-lined descriptions, after a `description:`, add one space, then insert `|-`. This escapes the YAML formatting and enables the use of Markdown.

## Avoid knowledge bias

Do not assume that the reader has the same knowledge of the product as you. We all do this to some extent. This is a cognitive bias known as the [Curse of knowledge](https://effectiviology.com/curse-of-knowledge).

> The curse of knowledge is a cognitive bias that occurs when an individual, communicating with other individuals, unknowingly assumes that the others have the background to understand.

Avoid this bias by describing information that may seem obvious to you when writing. For example, when describing `organizationId`, describe what that value is in a Rebilly context, or link to detailed information.

### Undesired example of the knowledge bias

```yaml
  organizationId:
    description: Organization ID.
```

> To understand what this ID value is, the reader must understand what an organization is in Rebilly. This is obvious to us because we have expert knowledge of the product. The reader may not.

Reviewer: ```suggestion
To understand what this ID value is, the reader must understand what an organization is in Rebilly. This may be obvious to you if you have expert knowledge of the product. The reader may not.
```

```yaml
summary: Retrieve a payment instrument
```

Deactivate a payment instrument:

```yaml
summary: Deactivate a payment instrument
```

## Descriptions

This section provides guidance on how to write operation, object, and parameter descriptions.

### Operations

- Start all operations with an active verb (Retrieves, Adds, Creates, Updates, Deletes, Sets, Specifies ...)
- Do not start with an imperative (Create, Retrieve, Merge ...). The user may read this as a command.
- Avoid starting with "The" or "A". Omit articles for ease of reading.
- Provide detail and link to related content if necessary.

:::success Tip

Place this imaginary text before your operation descriptions: _This operation..._.

Example: _"This operation"_ Creates a new customer account. \
Result: Creates a new customer account.

:::

#### Operation examples

Get a customer by ID:

```yaml
  operationId: GetCustomer
  description: |-
    Retrieves customer information based on the customer's ID.
```

Create a new customer:

```yaml
  operationId: PostCustomer
  description: Creates a new customer account.
```

Delete a customer by ID:

```yaml
  operationId: DeleteCustomer
  description: |-

>     Deletes a customer account based on the customer's `id`.

Reviewer: ```suggestion
    Deletes a customer account based on the customer's ID.
```
